File: Examples/ExampleOC/ExampleOC/ViewController.m
//
//  ViewController.m
//  ExampleOC
//
//  Created by CoderChan on 10/31/25.
//

#import "ViewController.h"
#import "MarsLog.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [MarsLogger.sharedLogger logWithLevel:(MarsLogLevelInfo) moduleName:@"1234" fileName:__FILE__ lineNumber:__LINE__ funcName:__FUNCTION__ format:@"viewDidLoad"];
}


@end

File: Examples/ExampleSwift/ExampleSwift/ViewController.swift
//
//  ViewController.swift
//  ExampleSwift
//
//  Created by CoderChan on 10/31/25.
//

import UIKit
import MarsLog

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        AppLog.log("viewDidLoad")
    }


}

struct AppLog {
    static func log(_ message: String, file: String = #file, line: Int32 = #line, function: String = #function) {
        let fileName = URL(fileURLWithPath: file).lastPathComponent
        
        MarsLogger.shared().log(with: .info, moduleName: "999", fileName: file, lineNumber: line, funcName: function, message: message)
    }
}
